### PR TITLE
tord binary search tree oop

### PR DESCRIPTION
**Pull request by Kazads**

### CODE REVIEW

File: OOBinarySearchTreeTord/es6-test/bst_test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import BinarySearchTree from '../dist/js/BinarySearchTree';
+import { expect } from 'chai';
+
+describe('BinarySearchTree', function() {
```
Comment (KimSvenSand) [inline OOBinarySearchTreeTord/es6-test/bst_test.js]: You're not using our test data. Make sure that insert works correctly without using untested functions.

Comment (KimSvenSand) [inline OOBinarySearchTreeTord/es6-test/bst_test.js:4]: You're not using our test-date. Make sure to test for example insert without using untested functions.

File: OOBinarySearchTreeTord/src/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  while(inserted < 10){
+    if(BST.insert(Math.floor((Math.random()*100)+1))){
+      inserted++;
+    }
+
+  }
+  //console.log(BST);
+  console.log(BST.inOrderTraversal());
+
+  let t0 = new Date();
```
Comment (KimSvenSand) [inline OOBinarySearchTreeTord/src/index.js:25]: You're not measuring the time of anything.

File: OOBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+export default class BinarySearchTree {
+  constructor(tree) {
+    if(tree.length > 0){
+      this.root = new _BinaryNode(tree[0]);
+      for(var i =1; i < tree.length; i++){
+        var node = this.root;
+        while(node != null){
```
Comment (KimSvenSand) [inline OOBinarySearchTreeTord/src/js/BinarySearchTree.js:9]: maybe use your insert function here?

File: OOBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      if(key < node.key){
+        node = node.leftNode;
+      }else if(key > node.key){
+        node = node.rightNode;
+      }
+    }
+    if(node != null){
+      returnValue = node.key;
+    }
+
```
Comment (KimSvenSand) [inline OOBinarySearchTreeTord/src/js/BinarySearchTree.js:45]: Return Node instead of key, since key is the input.

**Review by KimSvenSand — CHANGES_REQUESTED** [review]
Some small stuff, and we will have to test this with the new tests as well :)

File: OOBinarySearchTreeTord/src/js/BinarySearchTree.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+export default class BinarySearchTree {
+  constructor(tree) {
```
Comment (emilfolino) [inline OOBinarySearchTreeTord/src/js/BinarySearchTree.js:4]: Dela upp instantiering och skapande av träd? Och använd insert, är mer DRY.

**Review by emilfolino — COMMENTED** [review]
Bra implementation.

**Pull request closed without merging**